torchvision/csrc/io/image/cpu/exif.h

```diff
@@ -138,51 +182,48 @@ inline int fetch_exif_orientation(j_decompress_ptr cinfo) {
     cmarker = cmarker->next;
   }
 
-  if (exif_marker) {
-    // Exif binary structure looks like this
-    // First 6 bytes: [E, x, i, f, 0, 0]
-    // Endianness, 2 bytes : [M, M] or [I, I]
-    // Tag mark, 2 bytes: [0, 0x2a]
-    // Offset, 4 bytes
-    // Num entries, 2 bytes
-    // Tag entries and data, tag has 2 bytes and its data has 10 bytes
-    // For more details:
-    // http://www.media.mit.edu/pia/Research/deepview/exif.html
-
-    // Bytes from Exif size field to the first TIFF header
-    constexpr size_t start_offset = 6;
-    if (exif_marker->data_length > start_offset) {
-      auto* exif_data_ptr = exif_marker->data + start_offset;
-      auto size = exif_marker->data_length - start_offset;
-
-      ExifDataReader exif_data(exif_data_ptr, size);
-      auto endianness = get_endianness(exif_data);
-
-      // Checking whether Tag Mark (0x002A) correspond to one contained in the
-      // Jpeg file
-      uint16_t tag_mark = get_uint16(exif_data, endianness, 2);
-      if (tag_mark == REQ_EXIF_TAG_MARK) {
-        auto offset = get_uint32(exif_data, endianness, 4);
-        size_t num_entry = get_uint16(exif_data, endianness, offset);
-        offset += 2; // go to start of tag fields
-        constexpr size_t tiff_field_size = 12;
-        for (size_t entry = 0; entry < num_entry; entry++) {
-          // Here we just search for orientation tag and parse it
-          auto tag_num = get_uint16(exif_data, endianness, offset);
-          if (tag_num == INCORRECT_TAG) {
-            break;
-          }
-          if (tag_num == ORIENTATION_EXIF_TAG) {
-            exif_orientation = get_uint16(exif_data, endianness, offset + 8);
-            break;
-          }
-          offset += tiff_field_size;
-        }
-      }
-    }
+  if (!exif_marker) {
+    return -1;
   }
-  return exif_orientation;
+
+  constexpr size_t start_offset = 6;
+  if (exif_marker->data_length <= start_offset) {
+    return -1;
+  }
+
+  auto* exif_data_ptr = exif_marker->data + start_offset;
+  auto size = exif_marker->data_length - start_offset;
+
+  return fetch_exif_orientation(exif_data_ptr, size);
+}
+#else
+inline int fetch_jpeg_exif_orientation(j_decompress_ptr cinfo) {
+  return -1;
+}
+#endif // #if JPEG_FOUND
+
+#if PNG_FOUND
```

Comment:
Could we be using and / or logic on the pre-processor directives here? Something like
```suggestion
#if PNG_FOUND && ifdef PNG_eXIf_SUPPORTED
```

might make the logic a bit easier to parse (since we return -1 in both cases)